docs/templates.md

```diff
@@ -1,6 +1,36 @@
 # Headers and Decorators
 
 ## Template
+### Templates using decorators
+#### The usage
+```python
+from pyccel.decorators import template
+@template(name='T', types=['int','real'])
+def f(a : 'T', b : 'T'):
+	pass
+```
+Arguments:
+-   name: the name of the template
+-   types: the types the template represents.
+---
+*Note:*
+The arguments **name** and **types** could also be passed of the form
+`@template(T, types=['int', 'real'])` without specifying the keywords.and the order in which the decorators are called is not important.
```

Comment:
Should `T` be a string here?

Maybe the second sentence should be a separate point in a list, and not start with "and".

Reply:
Fixed 9934fdf8

I think the second point was about mixing `@types` and `@template`. I have removed it